Blog Post: Seventeenth Wave of Nostr Grants (#767)
OpenSats/website: data/blog/seventeenth-wave-of-nostr-grants.mdx

---
title: "Seventeenth Wave of Nostr Grants"

> date: '2026-05-19'

Reviewer: ```suggestion
date: '2026-05-20'
```

access to hosted or local inference infrastructure.

With support from this grant, Routstr will focus on hardening its minimum viable
protocol into a v1 release, improving reliability around edge cases, failures,
retries, and timeouts, and adding security and abuse-resistance improvements.
The roadmap also includes dynamic pricing, batch processing, performance
benchmarks, specialized worker-node software for local inference, better
operator documentation, [Umbrel][umbrel]-ready distribution, and application
work across Routstr Chat, agentic AI features, nostr feed ranking, and a
framework for building nostr bots.

Repositories: [Routstr][routstr-repo]  
License: GPL-3.0

[routstr-site]: https://routstr.com/
[ecash-topic]: /topics/ecash
[umbrel]: https://github.com/getumbrel/umbrel
[routstr-repo]: https://github.com/Routstr

---

### Wisp

[Wisp][wisp-site] is a nostr client for Android, built for speed and for users
new to the protocol. It implements the [outbox model][outbox-topic-wisp] and
supports dedicated DM relay sets for private messaging. The client includes a
non-custodial Lightning wallet built using [Breez SDK (Spark)][breez-sdk-spark]

> that runs as a node on the device, with seed-phrase recovery and encrypted

Reviewer 2: ```suggestion
non-custodial wallet based on the [Breez SDK (Spark)][breez-sdk-spark],
with seed-phrase recovery and encrypted
```

Got second thoughts on this. Technically speaking, the Breeze implementation is called "Nodeless" since it does not run a node on the device, but works through the Spark statechain.

So, I'd remove both the reference to `runs as a node on the device` and `Lightning` one, since it is now really a lightning wallet.

backup to the user's own relays. Upon signing up, a new user can receive
[zaps][zaps-topic], open a [private message][private-dms-topic] thread, and
upload photos through [Blossom][blossom-topic]. Wisp has been one of the
fastest-growing nostr clients on Android since launch.

With support from this grant, creator Barry Deen will work on the client
full-time. Near-term development includes a web client and a "normie mode" that
displays Lightning wallet balances in USD and lets users cash out with
[Bitrefill][bitrefill] gift cards. iOS development and beta launch are planned
for the second half of the year, along with private relay-based communities.

Repository: [barrydeen/wisp][wisp-repo]  
License: MIT

[wisp-site]: https://wisp.mobile
[outbox-topic-wisp]: /topics/outbox
[breez-sdk-spark]: https://github.com/breez/spark-sdk#breez-sdk---spark
[zaps-topic]: /topics/zaps
[private-dms-topic]: /topics/private-dms
[blossom-topic]: /topics/blossom
[bitrefill]: https://www.bitrefill.com/us/en/
[wisp-repo]: https://github.com/barrydeen/wisp

---

> Renewed support for Zapstore continues the work of building permissionless

Reviewer: ```suggestion
Renewed support for [Zapstore](/projects/zapstore) continues the work of building permissionless
```

OpenAI-compatible API endpoint: a client points an existing SDK or application
at a Routstr provider, pays per request with a Cashu token, and receives the
model response. This gives nostr applications a path to use AI services through
open discovery and bitcoin-native payments, while giving operators a way to sell
access to hosted or local inference infrastructure.

With support from this grant, Routstr will focus on hardening its minimum viable
protocol into a v1 release, improving reliability around edge cases, failures,
retries, and timeouts, and adding security and abuse-resistance improvements.
The roadmap also includes dynamic pricing, batch processing, performance
benchmarks, specialized worker-node software for local inference, better
operator documentation, [Umbrel][umbrel]-ready distribution, and application
work across Routstr Chat, agentic AI features, nostr feed ranking, and a
framework for building nostr bots.

Repositories: [Routstr][routstr-repo]  
License: GPL-3.0

[routstr-site]: https://routstr.com/
[ecash-topic]: /topics/ecash
[umbrel]: https://github.com/getumbrel/umbrel
[routstr-repo]: https://github.com/Routstr

---

### Wisp

> [Wisp][wisp-site] is a [nostr][nostr-topic-wisp] client for Android, built for

Reviewer 2: We did not link ot `nostr` topic in the other paragraphs, I'd keep it consistent and remove this.

speed and for users new to the protocol. It implements the [outbox
model][outbox-topic-wisp] and supports dedicated DM relay sets for private

> messaging. The client includes a non-custodial [Lightning][sats-topic-wisp]

Reviewer 2: Not sure if it makes sense to link to `sats` topic. I'd remove the link and maybe think about creating a `Lightning` topic of its own.
